src/sentry/snuba/metrics/query_builder.py

```diff
@@ -518,54 +519,102 @@ def __init__(
         self._alias_to_metric_field = {field.alias: field for field in self._metrics_query.select}
 
     @staticmethod
-    def generate_snql_for_groupby_field(
-        metric_groupby_obj: MetricGroupByField,
+    def generate_snql_for_action_by_fields(
+        metric_action_by_field: MetricActionByField,
         use_case_id: UseCaseKey,
         org_id: int,
         projects: Sequence[Project],
         is_column: bool = False,
-    ) -> Union[Column, AliasedExpression, Function]:
-        if isinstance(metric_groupby_obj.field, str):
-            if metric_groupby_obj.field == "transaction":
+    ) -> Union[List[OrderBy], Column, AliasedExpression, Function]:
+        """
+        Generates the necessary snql for any action by field which in our case will be group by and order by. This
+        function has been designed to share as much logic as possible, however, it should be refactored in case
+        the snql generation starts to diverge significantly.
+        """
+
+        is_group_by = isinstance(metric_action_by_field, MetricGroupByField)
+        is_order_by = isinstance(metric_action_by_field, MetricOrderByField)
+        if not is_group_by and not is_order_by:
```

Comment:
you can rewrite it with:
```
isinstance(metric_action_by_field, (MetricGroupByField, MetricOrderByField,))
```
https://docs.python.org/3/library/functions.html#isinstance

>If classinfo is neither a class object nor a type object, it may be a tuple of class or type objects, or may recursively contain other such tuples (other sequence types are not accepted).

Reply:
oh nvm, i see later you need it separately.

Reply:
The only problem with the use of booleans here is that the linter doesn't infer the automatic type cast but it is not a big deal because this approach improves readability